include/secp256k1_silentpayments.h

```diff
@@ -28,7 +28,41 @@ extern "C" {
  * operations.
  */
 
-/* TODO: add function API for sender side. */
+/** Create Silent Payment tweak data from input private keys.
+ *
+ * Given a list of n private keys a_1...a_n (one for each silent payment
+ * eligible input to spend) and a serialized outpoint_smallest, compute
+ * the corresponding input private keys tweak data:
+ *
+ * a_tweaked = (a_1 + a_2 + ... + a_n) * hash(outpoint_smallest || A)
+ *
+ * (where A = A_1 + A_2 + ... + A_n)
+ *
+ * If necessary, the private keys are negated to enforce the right y-parity.
+ * For that reason, the private keys have to be passed in via two different parameter
+ * pairs, depending on whether they were used for creating taproot outputs or not.
+ * The resulting data is needed to create a shared secret for the sender side.
+ *
+ *  Returns: 1 if shared secret creation was successful. 0 if an error occured.
+ *  Args:                  ctx: pointer to a context object
+ *  Out:  private_tweak_data32: pointer to the resulting 32-byte private tweak data
+ *  In:          plain_seckeys: pointer to an array of 32-byte private keys of non-taproot inputs
+ *                              (can be NULL if no private keys of non-taproot inputs are used)
+ *             n_plain_seckeys: the number of sender's non-taproot input private keys
+ *             taproot_seckeys: pointer to an array of 32-byte private keys of taproot inputs
+ *                              (can be NULL if no private keys of taproot inputs are used)
+ *           n_taproot_seckeys: the number of sender's taproot input private keys
+ *         outpoint_smallest36: serialized smallest outpoint
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_create_private_tweak_data(
+    const secp256k1_context *ctx,
+    unsigned char *private_tweak_data32,
+    const unsigned char *plain_seckeys,
```

Comment:
Started trying to use the module in #28122 and noticed this is an array of concatenated bytes, where `n` determines the number of items in the "list." Any particular reason for choosing this method of passing lists of pubkey/seckeys? Is this a convention in `libsecp`?

Reply:
> It seems a list of pointers (ending up in a double-pointer) makes more sense?

This makes more sense to me. If we have to concatenate the bytes for the secret keys, it seems like we would have to copy them out of secure memory just to pass them in to the function, whereas if we have a list of pointers the keys can be read directly from their secure memory.

> Should I change the interface to list of pointers? That seems strictly better than the current interface. Or is there another method of passing in key lists that you could think of?

list of pointers was going to be my suggestion :smile: 

> Oh btw, I found an old Bitcoin Core branch from December

Awesome, I'll take a look!